docs/routeCurrent.md

```diff
@@ -28,51 +53,59 @@ Or on its own:
 
 @signature `{{# routeCurrent([subsetMatch,] hashes...) }}FN{{else}}INVERSE{{/ routeCurrent }}`
 
-Renders `FN` if the `hashes` passed to [can-route.isCurrent route.isCurrent] returns `true`.
-Renders the `INVERSE` if [can-route.isCurrent route.isCurrent] returns `false`.
-
-```html
-<a class="{{# routeCurrent(true, page='todos') }}active{{/ routeCurrent }}">Todos</a>
-```
+  Renders `FN` if the `hashes` passed to [can-route.isCurrent route.isCurrent] returns `true`.
+  Renders the `INVERSE` if [can-route.isCurrent route.isCurrent] returns `false`.
+
+  ```html
+  <mock-url></mock-url>
+  <todo-item></todo-item>
+  <style>
+  .active{ background-color: red; }
+  .inactive{ background-color: dimgray; }
+  </style>
+  <script type="module">
+  import {Component, route} from "can";
+  import "//unpkg.com/mock-url@^5";
+
+  Component.extend({
+    tag: "todo-item",
+    view: `<a href='{{ routeUrl(page='todos' id=3) }}'
+      class='{{# routeCurrent(true, page='todos') }}
+        active
+      {{else}}
+        inactive
+      {{/ routeCurrent }}'>
+        {{todo.name}}
+      </a>`,
+    ViewModel: {
+      todo: {
+        default() {
+          return {name: "eat apple pie"};
+        }
+      }
+    }
+  });
+  </script>
+  ```
+  @codepen
 
   @param {Boolean} [subsetMatch] If an optional `true` is passed, `routeCurrent` will
   return `true` if every value in `hashes` matches the current route data, even if
   the route data has additional properties that are not matched.
 
   @param {can-stache/expressions/hash} hashes A hash expression like `page='edit' recipeId=id`.
 
-
-
   @param {can-stache.sectionRenderer} FN A subsection that will be rendered if the current route matches `hashes`.
 
   @param {can-stache.sectionRenderer} [INVERSE] An optional subsection that will be rendered
   if the current route does not match `hashes`.
 
   @return {String} The result of `SUBEXPRESSION` or `{{else}}` expression.
 
-
-
 @body
 
 ## Use
 
-Use the `routeCurrent` helper like:
-
-```html
-<li {{# routeCurrent(page="recipe" id=5) }}class='active'{{/ routeCurrent }}>
-  <a href='{{ routeUrl(page="recipe" id=5) }}'>{{recipe.name}}</a>
-</li>
-```
-
-With default routes and a url like `#!&page=recipe&id=5`, this produces:
-
-```html
-<li class='active'>
-  <a href='#!&page=recipe&id=5'>{{recipe.name}}</a>
-</li>
-```
-
-
 The following demo uses `routeCurrent` and [can-stache-route-helpers.routeUrl] to
 create links that update [can-route]’s `page` attribute:
 
```

Comment:
The demo should be codepen-able ... or remove the demo and just make it source that's codepened

Reply:
I changed the demo to follow the modern way we do things.

```html
<my-nav></my-nav>
<script type="module">
import {Component, route} from "can";

Component.extend({
  tag: "my-nav",
  view: `
    <a {{^ routeCurrent(page='home') }}
      href="{{ routeUrl(page='home') }}"
      {{/ routeCurrent }}
    >home</a>
    <a {{^ routeCurrent(page='restaurants') }}
      href="{{ routeUrl(page='restaurants') }}"
      {{/ routeCurrent }}
    >restaurants</a>
    {{# eq(routeData.page, 'home') }}
      <h1>Home page</h1>
    {{ else }}
      <h1>Restaurants page</h1>
    {{/ eq }}
  `,
  ViewModel: {
    routeData: {
      default() {
        route.register("{page}", {page: "home"});
        route.start();
        return route.data
      }
    }
  }
});
</script>
```